### ISSUE TITLE
RenderType does not apply when a vanilla model is used

### ISSUE THREAD

**Issue by Andromander** [MinecraftForge/MinecraftForge#8935]
**Minecraft Version:** 1.19.1

**Forge Version:** 42.0.1

**Logs:** n/a

**Steps to Reproduce:**
 1. Create a resource pack that applies to a block with no RenderType, or register a block that will reference a model in vanilla that has a RenderType registered to it. In this example, the resource pack will replace Stone Slab (the models are used for a newly registered block). 
 2. For new models, apply the RenderType in the model json itself. For the existing model, reference the vanilla model. In this example, the new models are a bottom and top slab with `minecraft:cutout_mipped`, and the existing model is a full block featuring Grass Block.
 3. Place the bottom, top, and full blocks in game.

**Description of issue:**
[Blockstate and Models in the reproduction step are here.](https://gist.github.com/Andromander/850b418815e07804253301c270a53be4)

The new models for the bottom and top slab render with `cutout_mipped`, but the model for the full block using Grass Block renders as if it was `solid` when the intended result is to render with `cutout_mipped`.
![2022-08-02_10 59 22](https://user-images.githubusercontent.com/29526717/182270588-03afddf4-4ba4-44ab-bc28-b4f604992593.png)
My guess is that the block itself is rendered as `cutout_mipped` thanks to Vanilla, but if the model is used outside of the block, the RenderType no longer applies. There's not really a way to render it correctly without either:
1. Register the RenderType through ItemBlockRenderTypes (this method is marked for removal).
2. Overriding the vanilla model file in the mod (this could conflict with resource packs, probably not an ideal solution, either).
3. Override `IForgeBakedModel#getRenderTypes` (most viable, but a hassle for a fairly small task, and could still impact resource packs).

Ideally, it would be better if it were possible to reference a Vanilla block model and the RenderType applied to the block was also applied to the models if no RenderType is explicitly stated, without having to make new models.

**Comment by gigaherz** [comment]
You should be able to fix that by having your own separate model with
```
{
"parent": "minecraft:block/grass",
"render_type": "cutout_mipped"
}
```
This would let you avoid the "conflict with resource packs" issue.

**Comment by amadornes** [comment]
I intended to post this sooner, but as discussed on Discord, the reason this doesn't work is that render types for vanilla blocks are hardcoded, and as such that information is not available in the model JSONs.

Inferring a render type from vanilla models like you want isn't feasible for two main reasons:
1. There is nothing linking a model JSON with a block, so the in-code mappings cannot be retrieved, and
2. Even if they could be, the same model JSON could be used by multiple blocks with different render types (think resourcepacks) so there is no guarantee that there wouldn't be any collisions

Your best bet would be to do what we discussed on Discord and giga posted above: just create a new model that derives from the one you're targetting, and specify the render type there.
